src/Program.cs

```diff
@@ -44,18 +44,21 @@ private static byte[] ComputeSignature(Archive archive)
         {
             var taxSummary = archive.TaxSummary;
             var reportedValue = archive.ReportedValue;
-            var operationName = "ARCHIVING";
             var previousSignatureFlag = archive.Metadata.PreviousRecordSignature.Match(
                 _ => "Y",
                 _ => "N"
             );
+            var archiveType = archive.Metadata.Version.Match(
+                ArchiveVersion.v100, _ => "ARCHIVING",
+                ArchiveVersion.v400, _ => archive.Metadata.ArchiveType.GetOrElse("")
```

Comment:
The whole app is written in a way that invalid data lead to a different failure than the actual signature check failure. I would follow this principle here as well and make sure model metadata doesn't even get created if it's missing some information.